app/scripts/migrations/135.ts

```diff
@@ -0,0 +1,147 @@
+import { hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 135;
+
+/**
+ * Removes tokens with `decimals === null` from `allTokens`, `allIgnoredTokens`, and `allDetectedTokens`.
```

Comment:
We probably want to migrate `tokens` and `detectedTokens` too.  These are the variants that are scoped to the current chain and account.  Otherwise the migrated state may not get swapped into `tokens` and `detectedTokens` until a chain or account switch occurs.

We should eventually deprecate them in favor of the `all*` variants.  But in the meantime I think they still may be queried by the UI.

Reply:
will do it now

Reply:
pushed here [26d31ad](https://github.com/MetaMask/metamask-extension/pull/29245/commits/26d31ad2c49b8f91023fe4eeef97807737a4abdc)
will do some manual tests before submitting for review